video/out/wayland_common.c

```diff
@@ -1832,6 +1828,14 @@ int vo_wayland_init(struct vo *vo)
                  xdg_wm_base_interface.name);
         return false;
     }
+    if (wl->subcompositor) {
+        wl->video_subsurface = wl_subcompositor_get_subsurface(wl->subcompositor, wl->video_surface, wl->surface);
+        wl_subsurface_set_desync (wl->video_subsurface);
+    }
+    if (wl->viewporter) {
+        wl->viewport = wp_viewporter_get_viewport (wl->viewporter, wl->surface);
+        wl->video_viewport = wp_viewporter_get_viewport (wl->viewporter, wl->video_surface);
+    }
```

Comment:
Could you move these so they occur after the `if (create_xdg_surface(wl))` part? Also no spaces between the parenthesis and the function.